senator/game.py

```diff
@@ -11,22 +11,33 @@ def regret_matching(regret: np.ndarray) -> np.ndarray:
 
 
 class Node:
-    def __init__(self, parent, utility_matrix):
+    def __init__(self, parent, utility_matrix: np.ndarray) -> None:
+
         self.parent = parent
         self.utility_matrix = utility_matrix
         self.iterations = 0
 
         self._num_players = utility_matrix.shape[0]
         self._num_resources = utility_matrix.shape[1]
+
         self._cumulative_regret = np.zeros((self._num_players, self._num_resources))
-        self._cumulative_positive_regret = np.zeros((self._num_players, self._num_resources))
-        self.children = {}  # key: action, value: Node
+        self._cumulative_strategy = np.zeros((self._num_players, self._num_resources))
+
+        self.children: dict[tuple, Node] = {}  # key: action, value: Node
 
     @property
     def current_strategy(self) -> np.ndarray:
-        strategy = regret_matching(self._cumulative_positive_regret)
+        strategy = regret_matching(self._cumulative_regret)
         return strategy
 
+    @property
+    def average_strategy(self) -> np.ndarray:
+        strategy = regret_matching(self._cumulative_strategy)
+
+        strategy_sum = strategy.sum(axis=1)
```

Comment:
Tohle není nikdy použitý. Navíc to není třeba, `regret_matching` už zajistí to, že je `strategy` správně normalizovaná. Takže můžeš prostě vrátit strategy a čus

Reply:
To přesně dělá current_strategy. Na co jsem měl teda udělat ještě average?

Reply:
Ok, už jsme se pochopili